CRM/Contact/Import/Form/MapField.php

```diff
@@ -595,4 +595,31 @@ private function isLocationTypeRequired($name): bool {
     return (bool) (Civi::$statics[__CLASS__]['location_fields'][$name] ?? FALSE);
   }
 
+  /**
+   * Assign js to hide hide unused fields.
+   *
+   * The mapper fields use QuickForm hierarchical select and, depending on the
+   * first field selection may offer more (hierarchical) fields. Eg. if you
+   * choose 'phone' for the first field then 2 more fields will give you
+   * the chance to enter the phone_type_id and location_type_id.
+   *
+   * When loading the field we hide these hierarchical fields if they are empty,
+   * by js which is assigned here.
+   *
+   * @param array $defaults
+   */
+  protected function assignJsToHideUnusedFields(array $defaults): void {
+    $js = ["<script type='text/javascript'>"];
+    foreach ($defaults as $row => $mapperValues) {
+      foreach ([1, 2, 3] as $columnNumber) {
+        if (empty($mapperValues[$columnNumber])) {
+          $rowNumber = str_replace('mapper[', '', substr($row, 0, -1));
+          $js[] = "cj('#mapper_{$rowNumber}_{$columnNumber}').hide();";
```

Comment:
Isn't 'cj' deprecated?  How about:
`$js[] = "CRM.\$('#mapper_{$rowNumber}_{$columnNumber}').hide();";`

Reply:
yeah - I'm always confused about that - it doesn't seem to work on some pages  - @colemanw what should I use (Or are we just happy with 'a bit better' here?)